vmdk_plugin/shared_plugin/main.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+// A Shared Docker Data Volume plugin - main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"reflect"
+	"runtime"
+	"syscall"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/go-plugins-helpers/volume"
+	"github.com/vmware/docker-volume-vsphere/vmdk_plugin/drivers/shared"
+	"github.com/vmware/docker-volume-vsphere/vmdk_plugin/utils/config"
+	"github.com/vmware/docker-volume-vsphere/vmdk_plugin/utils/log_formatter"
+	"github.com/vmware/docker-volume-vsphere/vmdk_plugin/utils/plugin_server"
+)
+
+// main for docker-volume-vsphere
+// Parses flags, initializes and mounts refcounters and finally initializes the server.
```

Comment:
looks like the majority of the code is redundant in both drivers and should be separated in main_utils.go :-)